pkg/resources/management.cattle.io/v3/globalrolebinding/validator_test.go

```diff
@@ -241,11 +254,48 @@ func TestAdmit(t *testing.T) {
 					return baseGRB
 				},
 				oldGRB: func() *v3.GlobalRoleBinding { return nil },
+				stateSetup: func(ts testState) {
+					setSarResponse(false, nil, testUser, adminGR.Name, ts.sarMock)
+				},
+			},
+			allowed: false,
+		},
+		{
+			name: "privilege escalation self, bind allowed",
+			args: args{
+				username: testUser,
+				newGRB: func() *v3.GlobalRoleBinding {
+					baseGRB := newDefaultGRB()
+					baseGRB.UserName = testUser
+					baseGRB.GlobalRoleName = adminGR.Name
+					return baseGRB
+				},
+				oldGRB: func() *v3.GlobalRoleBinding { return nil },
+				stateSetup: func(ts testState) {
+					setSarResponse(true, nil, testUser, adminGR.Name, ts.sarMock)
+				},
+			},
+			allowed: true,
+		},
+		{
```

Comment:
For tests in general I think it would be great to verify that it works correctly when:
- resourceNames == "*"
- resourceNames == list of known values (minimally 1 value and 2 values)

I also wonder if we should have an integration test that covers the use case we're trying to allow: granting both escalate and bind to an SA so that it can create GR/GRBs that it previously couldn't. (Mostly for regression testing)

Reply:
Discussed elsewhere, putting the conclusion here:
> I'm not sure what you mean here. ...
I was talking about the `resourceNames` in the CR.

Both comments should have been about integration tests, not unit test.

We're okay with no integration test for this, considering the above. QA could potentially automate the testing of this usecase, unclear if will happen.